contracts/container/contract.go

```diff
@@ -238,6 +239,42 @@ func Update(script []byte, manifest []byte, data any) {
 	runtime.Log("container contract updated")
 }
 
+// PutObject registers successful object PUT operation and notifies about
+// it. metaInformation must be signed by container nodes according to
+// container's placement, see [VerifyPlacementSignatures]. metaInformation
+// must contain information about an object placed to a container that was
+// created with meta-on-chain feature enabled.
+func PutObject(metaInformation []byte, sigs [][]interop.Signature) {
+	metaMap := std.Deserialize(metaInformation).(map[string]any)
+	cID := metaMap["cid"].(interop.Hash256)
+	if len(cID) != interop.Hash256Len {
+		panic("incorrect container ID")
+	}
+
+	if storage.Get(storage.GetContext(), append([]byte{containersWithMetaPrefix}, cID...)) == nil {
+		panic("container does not support meta-on-chain")
+	}
+
+	if !VerifyPlacementSignatures(cID, metaInformation, sigs) {
+		panic("signature verification failed")
+	}
+
+	oid := metaMap["oid"].(interop.Hash256)
+	runtime.Notify("ObjectPut", cID, oid)
+}
+
+// PutMeta is the same as [Put] (and exposed as put from the contract via
+// overload), but allows container's meta-information be handled and notified
+// using the chain.
+func PutMeta(container []byte, signature interop.Signature, publicKey interop.PublicKey, token []byte, metaOnChain bool) {
```

Comment:
we put container and just configure meta support, with this `PutMeta` name can confuse. I can suggest `PutWithTunedMeta`, but not very against ur proposal

Reply:
i have adopted @roman-khimov's [approach](https://github.com/nspcc-dev/neofs-contract/blob/f6398fe878a1a4285f522ca2d433781812c6bba4/contracts/netmap/contract.go#L497) with method overloading. no one will call it `putMeta`, it is the same `put` but with one more arg. i really liked it

Reply:
`put` but with one more arg - `putWithMeta` to me